### PR TITLE
[Hooks] Add: Plugin activation and deactivation hooks

### PR DESCRIPTION
**Pull request by Chhekur**

### CODE REVIEW

**Review by piyush-getsimpl — APPROVED** [review]

**Comment by piyush-getsimpl** [comment]
Please resolve conflicts.

**Comment by Chhekur** [comment]
> Please resolve conflicts.

done .

File: includes/hook_handlers/plugin-status-hook-handler.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    $request["event"] = "updated";
+    $data = array();
+    $data["plugin_status"] = $plugin_status;
+    $request["data"] = $data;
+
+    $checkout_3pp_client = new SimplCheckout3ppClient();
+    try {
+        $simplHttpResponse = $checkout_3pp_client->post_hook_request($request);
+    } catch (\Throwable $th) { 
+        error_log(print_r($th, TRUE));
```
Comment (abhaymaniyar) [inline includes/hook_handlers/plugin-status-hook-handler.php:14]: Let's use the simpl_logger to log this now.

File: includes/hook_handlers/plugin-status-hook-handler.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        error_log(print_r($th, TRUE)); 
+    }
+}
+
+function plugin_status_activate_hook() {
+    plugin_status_hook("active");
+}
+
+function plugin_status_deactivate_hook() {
+    plugin_status_hook("in_active");
```
Comment (abhaymaniyar) [inline includes/hook_handlers/plugin-status-hook-handler.php:23]:
```suggestion
    plugin_status_hook("inactive");
```

File: simpl-checkout-woocommerce.php [hunk 3]
```diff
@@ -5,7 +5,7 @@
  * Description: Simpl checkout offers an optimised checkout process, higher order conversions and RTO reduction. We offer Simpl Pay Later, Pay-in-3, UPI, Cards, and COD for seamless transactions while you focus on growing your business.
  * Author:  One Sigma Technologies Pvt. Ltd.
  * Author URI: http://www.getsimpl.com
- * Version: 1.2.5
+ * Version: 1.2.6
```
Comment (abhaymaniyar) [inline simpl-checkout-woocommerce.php:8]:
```suggestion
 * Version: 1.2.7
```

File: includes/hook_handlers/plugin-status-hook-handler.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?php
+function plugin_status_hook($plugin_status) {
+    $request["topic"] = "plugin_status.updated";
+    $request["resource"] = "plugin_status";
+    $request["event"] = "updated";
+    $data = array();
+    $data["plugin_status"] = $plugin_status;
```
Comment (abhaymaniyar) [inline includes/hook_handlers/plugin-status-hook-handler.php:7]:
```suggestion
    $data["status"] = $plugin_status;
```

File: includes/hook_handlers/plugin-status-hook-handler.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<?php
+function plugin_status_hook($plugin_status) {
+    $request["topic"] = "plugin_status.updated";
```
Comment (abhaymaniyar) [inline includes/hook_handlers/plugin-status-hook-handler.php:3]:
```suggestion
    $request["topic"] = "plugin.updated";
```

**Review by abhaymaniyar — APPROVED** [review]

**Pull request merged**